Assets/HoloToolkit/Common/Scripts/CrossPlatform/Reflection/ReflectionExtensions.cs

```diff
@@ -179,11 +179,12 @@ public static object[] GetCustomAttributes(this Type type, Type attributeType, b
 
     public static class ReflectionExtensions
     {
+#if !NET_4_6
```

Comment:
Does this define work? I'm still seeing the error in 2017.2.1f1.

![image](https://user-images.githubusercontent.com/3580640/36758287-dd0d0394-1bc8-11e8-9980-caa4ce4e4a7d.png)

Reply:
Restart the editor.

Reply:
https://docs.unity3d.com/Manual/PlatformDependentCompilation.html

Reply:
I know it exists in documentation, but I've never seen it work.

I just restarted the editor, deleted all the Unity generated files and folders, and I still see the error.  Which Unity version are you using?

Reply:
2017.2, 2017.1, and 2017.3

Reply:
Strange, I tried 
* 2017.3.1f1
* 2017.2.1p2
* 2017.2.1f1
* 2017.1.3f1
* 5.6.5f1

and 5.6 was the only one that didn't give me the error.

Reply:
What's stranger, is that I never get had the issue in the first place.  I wonder if it has something to do with what .NET workloads are installed on the machine.

Reply:
The ambiguous reference is exactly what we're trying to fix here.

Reply:
Are you sure you pulled in the fix?

Reply:
Aha! I went back and double checked everything, and I realized you've based this on the Stabilization branch. The issue isn't present there, only in the Dev_Working branch.

Reply:
Yeah the same will need to be addressed in the dev branch.  Although I'd like to see MRTK vNEXT use strictly use .NET 4.6.

Reply:
This issue isn't present in Stabilization at all, even without this change.
This fix doesn't fix the Dev branch, since it seems like the define doesn't work like it should. It seems the code isn't able to disambiguate between extension methods in different namespaces. I think the issue is that the Toolkit extensions are now in a namespace in the Dev branch, with a corresponding `using` which didn't used to be present.

Reply:
You're right.